### DISCUSSION TITLE
Skip welcome screen testing type and browser selection.

### DISCUSSION THREAD

**Discussion by tgreen7** [cypress-io/cypress#23051] [Questions and Help] [answered]
With the new version of cypress starting up leads to a page where you have to first select testing type (e2e or component) and then a browser. Is there a way to choose a default browser and testing type so that I am immediately launched into this page?

![Screen Shot 2022-08-01 at 3 30 56 PM](https://user-images.githubusercontent.com/10415101/182256493-586572c0-1b77-4e94-a113-8a6233f2c49c.png)

**Comment by liambutler** [comment, marked as answer]
Not currently a way to set this via the UI, but you can pass in commands like so:

```js
npx cypress open --e2e --browser electron  // goes directly into the spec view
npx cypress open --component --browser electron // does the same for component tests
```

**Reply by tgreen7** [reply]
This is perfect! Thanks

**Reply by phil294** [reply]
This still opens two separate windows though :/

**Comment by tnrich** [comment]
@liambutler love those command options! Is there any chance that we could get one like `--auto-start` that just starts running the first available spec in the list? 

Example: 
```
cypress open --auto-start --e2e --browser chrome --config specPattern="/Users/tnrich/Sites/openVectorEditor/cypress/e2e/editor.spec.js"
```

That would allow me to run tests with a single click in vscode and truly be a dream 



 🙏 Thank you 🙏

**Reply by liambutler** [reply]
I don’t work for Cypress. You would have more luck raising a feature request, or submitting a PR yourself

**Reply by lmiller1990** [reply]
I think we should probably remember the last used browser, and launch straight there. Then all you'd need is `yarn cypress open --e2e` and you'd be in the runner, skipping electron entirely.

Somewhat related: https://github.com/cypress-io/cypress/issues/21778

If you'd like to make an issue with this one specific improvement (default to previously used browser) that'd be great :+1:
